docs/content/guides/dagster/re-execution.mdx

```diff
@@ -0,0 +1,159 @@
+- **From Failure**: Re-execute the pipeline run, skipping steps completed successfully. This is only enabled when the pipeline has failed. You have likely fixed the failed step, and want to run the step and all downstream dependencies again. Dagster will figure out the dependencies for you!
+
+In the above example, re-executing from failure would make sense as the failed task has a 50% chance of succeeding on the next run.
+
+### Selecting Steps
+
+Within Dagit, a single or multiple steps may be selected simply by clicking them with the mouse. Alternatively, the text-box can be leveraged by typing step names to be re-run, adding a `+` for every subsequent step to be selected. This is demoed in the video below:
+
+<iframe
+  width="560"
+  height="315"
+  src="https://www.youtube.com/embed/LWc8MxSN22c"
+  frameBorder="0"
+  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+  allowfullscreen
+></iframe>
+
+## Relevant APIs
+
+Re-execution can be triggered via the API as well.
+
+| Name                                                                                       | Description                         |
+| ------------------------------------------------------------------------------------------ | ----------------------------------- |
+| <PyObject module="dagster" object="reexecute_pipeline" displayText="reexecute_pipeline" /> | Reexecute an existing pipeline run. |
+
+Consider the pipeline `unreliable_pipeline`, which has a solid named `unreliable`.
+
+```python file=../../reexecution/reexecution_api.py
+from dagster import DagsterInstance, execute_pipeline, reexecute_pipeline
+from reexecution.pipeline.unreliable_pipeline import unreliable_pipeline
+
+
+def reexecution_api_example():
+    instance = DagsterInstance.ephemeral()
+    run_config = {"intermediate_storage": {"filesystem": {}}}
+
+    # Initial execution
+    pipeline_result_full = execute_pipeline(unreliable_pipeline, run_config=run_config, instance=instance)
+
+    if not pipeline_result_full.success:
+        # Re-execution: Entire pipeline
+        reexecution_result_full = reexecute_pipeline(
+            unreliable_pipeline,
+            parent_run_id=pipeline_result_full.run_id,
+            run_config=run_config,
+            instance=instance,
+        )
+```
+
+Using Dagster's API, you can programmatically trigger both an execution and a reexecution. Upon an initial pipeline run failing, you may want to trigger a full re-execution. Similarly, you can trigger a re-execution of selected steps or from a particular point.
+
+```python file=../../reexecution/reexecution_api_steps.py
+# Re-execution: Starting with the "unreliable" solid and all its descendents
+reexecution_result_specific_selection = reexecute_pipeline(
+    unreliable_pipeline,
+    parent_run_id=pipeline_result_full.run_id,
+    run_config=run_config,
+    instance=instance,
+    solid_selection=['unreliable+*']
```

Comment:
We have a solid selection syntax, maybe it's worth mentioning that and link it to the api doc https://dagster-40z5pnwgn-elementl.vercel.app/_apidocs/execution#dagster.reexecute_pipeline so readers could be aware of that syntaxt.

Reply:
Ah yes, definitely makes sense to link. One question: I didn't see any API references to re-execute from failure. Did I miss it or is that not functionality currently present in the API?

Reply:
we don't have that functionality via API at the comment.